main.go

```diff
@@ -204,6 +208,7 @@ func NewPrometheus() *prometheus {
 		ConsolesHandler: consolesHandler,
 		AlertsHandler:   alertsHandler,
 		GraphsHandler:   graphsHandler,
+		Federation:      fed,
```

Comment:
Shouldn't this field and the type also be `...Handler` like the others? For consistency, but also because federation in itself is more of a concept, not a concrete thing...